### PR TITLE
Add details component

### PR DESCRIPTION
**Pull request by tijmenb**
The details component is useful to reveal more detailed information.

https://govuk-publishing-compon-pr-453.herokuapp.com/component-guide/details

https://trello.com/c/d1jqgiXk/54-add-details-component

### CODE REVIEW

File: app/assets/stylesheets/govuk_publishing_components/components/_details.scss [hunk 1]
```diff
@@ -0,0 +1 @@
+// Use the Details component from GOV.UK Frontend
```
Comment (andysellick) [inline app/assets/stylesheets/govuk_publishing_components/components/_details.scss]: Is this file necessary? I get what it's here to explain, but is that important?

File: app/views/govuk_publishing_components/components/docs/details.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+shared_accessibility_criteria:
+  - link
+accessibility_criteria: |
+  The component must:
+
+  * accept focus
+  * be focusable with a keyboard
+  * be usable with a keyboard
+  * be usable with touch
+  * indicate when it has focus
```
Comment (alex-ju) [inline app/views/govuk_publishing_components/components/docs/details.yml]: I would add:
* toggle the visibility of the details element's content when interacted with
* indicate the expanded state when details' content is visible
* indicate the collapsed state when details' content is hidden

File: app/assets/stylesheets/govuk_publishing_components/components/_details.scss (hunk 1 above)
Comment (tijmenb) [reply app/assets/stylesheets/govuk_publishing_components/components/_details.scss]: Yeah, we've done this for every component that uses GOV.UK Frontend. I think it's important so that devs can go to any component's CSS file and discover where the styling comes from. If we didn't have it, it would be up to the developer to find out where the styles come from.

File: app/views/govuk_publishing_components/components/docs/details.yml (hunk 2 above)
Comment (tijmenb) [reply app/views/govuk_publishing_components/components/docs/details.yml]: Thanks, updated.

Comment (alex-ju) [reply app/views/govuk_publishing_components/components/docs/details.yml]: Can't see them in yet. Happy to approve after they're added.

Comment (tijmenb) [reply app/views/govuk_publishing_components/components/docs/details.yml]: They're there 👇
